### PR TITLE
Get farmers endpoint

### PR DESCRIPTION
**Pull request by ayenisholah**
# Description

Describe change or new feature here.

## Checklist

Remove any items which are not applicable.

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works AND the tests pass

### CODE REVIEW

**Comment by ayenisholah** [comment]
[Create API endpoints to get all farmers and single farmer](https://trello.com/c/9V5VCZms/29-create-api-endpoints-to-get-all-farmers-and-single-farmer)

File: README.md [hunk 1]
```diff
@@ -39,6 +39,10 @@ To get the server running locally:
 | POST   | `/api/v1/user/signup`    | admin          | Create a new user account (staff and admin) | True                          |
 | POST   | `/api/v1/user/login`     | admin && staff | Login user (staff and admin)                | False                         |
 | POST   | `/api/v1/farmers/create` | admin && staff | Create new farmer                           | True                          |
+| GET    | `/api/v1/farmers` 				| admin && staff | Get All Farmers	                           | True                          |
+| GET    | `/api/v1/id`			 				| admin && staff | Get Farmer By Id	                           | True                          |
+
+
```
Comment (Rexben001) [inline README.md]: Can you change the API route  from `/api/v1/id` to `/api/v1/farmers/:id`

File: controllers/farmer/getFarmers.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ *
+ * @param {object} req
+ * @param {object} res
+ * @param {object} next
+ */
+
+const getAllFarmers = async (req, res, next) => {
+  try {
+    const farmers = await models.Farmer.find({});
+
```
Comment (Rexben001) [inline controllers/farmer/getFarmers.js]: You should check if the farmer record is empty and return a response

File: controllers/farmer/getFarmers.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        status: NOT_FOUND
+      })
+    );
+  }
+};
+
+const getFarmerById = async (req, res, next) => {
+  try {
+    const farmer = await models.Farmer.findById(req.params.id);
+
```
Comment (Rexben001) [inline controllers/farmer/getFarmers.js]: What if there's no farmer with that id?
Try to return a response.

File: middlewares/Validations/Users.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        });
+      } else {
+        req.body.username = req.body.username.replace(/\s+/g, '');
+        next();
+      }
+    });
+  }
+}
+
+module.exports = UserValidation;
```
Comment (Rexben001) [inline middlewares/Validations/Users.js]: This file is no more needed

File: controllers/farmer/getFarmers.js (hunk 2 above)
Comment (ayenisholah) [reply controllers/farmer/getFarmers.js]: Fixed.

File: README.md (hunk 1 above)
Comment (ayenisholah) [reply README.md]: Done

File: controllers/farmer/getFarmers.js (hunk 2 above)
Comment (ayenisholah) [reply controllers/farmer/getFarmers.js]: I need assistance with this one

File: README.md (hunk 1 above)
Comment (Rexben001) [reply README.md]: `farmers` and not `farmer`

File: controllers/farmer/getFarmers.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  try {
+    const { id } = req.params.id;
+    const farmer = await models.Farmer.findById(req.params.id);
+    if (id) {
+      return res.status(200).json({
+        success: true,
+        message: 'Farmer record found',
+        farmer
+      });
+    }
```
Comment (Rexben001) [inline controllers/farmer/getFarmers.js]: Can you explain what your `if` block is doing
You checking if `id` exists instead if farmer exists

File: routes/farmerRouter.js [hunk 6]
```diff
@@ -9,15 +9,17 @@ farmerRouter.post(
   '/farmers/create',
   verifyToken,
   validate.createFarmer,
-  FarmerController.addFarmer
+  farmerController.addFarmer
 );
 
+farmerRouter.get('/farmers', verifyToken, farmerController.getAllFarmers);
+farmerRouter.get('/farmers/:id', verifyToken, farmerController.getFarmerById);
```
Comment (Rexben001) [inline routes/farmerRouter.js]: Pls, add `validate.validateId` to validate the id

File: controllers/farmer/getFarmers.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        message: 'Farmers records not found',
+        status: NOT_FOUND
+      })
+    );
+  }
+};
+
+const getFarmerById = async (req, res, next) => {
+  try {
+    const { id } = req.params.id;
```
Comment (Rexben001) [inline controllers/farmer/getFarmers.js]: Can you explain this line?

File: tests/farmers.test.js [hunk 8]
```diff
@@ -69,7 +69,31 @@ describe('Farmer route', () => {
...
+        .get('/api/v1/farmers')
+        .set('Authorization', token)
+        .end((err, res) => {
+          res.should.have.status(200);
+          res.body.should.be.a('object');
+          res.body.message.should.equal('Farmers records found');
+          done(err);
+        });
+    });
+    it('It should return an array of farmers', (done) => {
```
Comment (Rexben001) [inline tests/farmers.test.js]: There's a typo.
`It should return an array of farmers` should be `It should return a single farmer`

Comment (ayenisholah) [reply tests/farmers.test.js]: Done

**Pull request merged**